integration_tests/CMakeLists.txt

```
@@ -355,6 +355,7 @@ RUN(NAME arrays_op_23 LABELS gfortran llvm)
 RUN(NAME arrays_op_24 LABELS gfortran llvm)
 RUN(NAME arrays_op_25 LABELS gfortran llvm)
 RUN(NAME arrays_reshape_14 LABELS gfortran llvm)
+RUN(NAME arrays_reshape_15 LABELS gfortran llvm)
```

Does this now work in the fortran backend?
```suggestion
RUN(NAME arrays_reshape_15 LABELS gfortran llvm fortran)
```

---

Good question. Do `--dump-all-passes` and see which pass adds the tanh. Also we should not call any passes before the Fortran backend, since we actually want to print all passes as fortran.

---

I think there's no track of `tanh`.

When I executed:
```bash
./src/bin/lfortran integration_tests/arrays_reshape_15.f90 --dump-all-passes
```
<details>
<summary>output</summary>

<p>

```python
  1.0
  3.0
  5.0

  2.0
  4.0
  6.0
```
</p>
</details>
After running:

```bash
./src/bin/lfortran integration_tests/arrays_reshape_15.f90 --backend=fortran --dump-all-passes
```

<details>
<summary>output</summary>

<p>

```python
Internal Compiler Error: Unhandled exception
Traceback (most recent call last):
  Binary file "/home/khushi/Documents/lfortran/src/bin/lfortran", in _start()
  File "./csu/../csu/libc-start.c", line 392, in __libc_start_main_impl()
  File "./csu/../sysdeps/nptl/libc_start_call_main.h", line 58, in __libc_start_call_main()
  File "/home/khushi/Documents/lfortran/src/bin/lfortran.cpp", line 2362, in ??
    err = compile_to_binary_fortran(arg_file, tmp_o, compiler_options);
  File "/home/khushi/Documents/lfortran/src/bin/lfortran.cpp", line 1456, in ??
    LCompilers::Result<std::string> src = fe.get_fortran(input, lm, diagnostics);
  File "/home/khushi/Documents/lfortran/src/lfortran/fortran_evaluator.cpp", line 538, in LCompilers::FortranEvaluator::get_fortran(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, LCompilers::LocationManager&, LCompilers::diag::Diagnostics&)
    return asr_to_fortran(*asr.result);
  File "/home/khushi/Documents/lfortran/src/libasr/codegen/asr_to_fortran.cpp", line 1139, in LCompilers::asr_to_fortran[abi:cxx11](LCompilers::ASR::TranslationUnit_t&, bool, int)
    v.visit_TranslationUnit(asr);
  File "/home/khushi/Documents/lfortran/src/libasr/codegen/asr_to_fortran.cpp", line 179, in LCompilers::ASRToFortranVisitor::visit_TranslationUnit(LCompilers::ASR::TranslationUnit_t const&)
    visit_symbol(*item.second);
  File "/home/khushi/Documents/lfortran/src/libasr/../libasr/asr.h", line 5060, in LCompilers::ASR::BaseVisitor<LCompilers::ASRToFortranVisitor>::visit_symbol(LCompilers::ASR::symbol_t const&)
    void visit_symbol(const symbol_t &b) { visit_symbol_t(b, self()); }
  File "/home/khushi/Documents/lfortran/src/libasr/../libasr/asr.h", line 4774, in ??
    case symbolType::Program: { v.visit_Program((const Program_t &)x); return; }
  File "/home/khushi/Documents/lfortran/src/libasr/codegen/asr_to_fortran.cpp", line 203, in LCompilers::ASRToFortranVisitor::visit_Program(LCompilers::ASR::Program_t const&)
    visit_body(x, r, false);
  File "/home/khushi/Documents/lfortran/src/libasr/codegen/asr_to_fortran.cpp", line 112, in void LCompilers::ASRToFortranVisitor::visit_body<LCompilers::ASR::Program_t>(LCompilers::ASR::Program_t const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, bool)
    visit_stmt(*x.m_body[i]);
  File "/home/khushi/Documents/lfortran/src/libasr/../libasr/asr.h", line 5077, in LCompilers::ASR::BaseVisitor<LCompilers::ASRToFortranVisitor>::visit_stmt(LCompilers::ASR::stmt_t const&)
    void visit_stmt(const stmt_t &b) { visit_stmt_t(b, self()); }
  File "/home/khushi/Documents/lfortran/src/libasr/../libasr/asr.h", line 4800, in ??
    case stmtType::Assignment: { v.visit_Assignment((const Assignment_t &)x); return; }
  File "/home/khushi/Documents/lfortran/src/libasr/codegen/asr_to_fortran.cpp", line 410, in LCompilers::ASRToFortranVisitor::visit_Assignment(LCompilers::ASR::Assignment_t const&)
    visit_expr(*x.m_value);
  File "/home/khushi/Documents/lfortran/src/libasr/../libasr/asr.h", line 5124, in LCompilers::ASR::BaseVisitor<LCompilers::ASRToFortranVisitor>::visit_expr(LCompilers::ASR::expr_t const&)
    void visit_expr(const expr_t &b) { visit_expr_t(b, self()); }
  File "/home/khushi/Documents/lfortran/src/libasr/../libasr/asr.h", line 4919, in ??
    case exprType::ArrayReshape: { v.visit_ArrayReshape((const ArrayReshape_t &)x); return; }
  File "/home/khushi/Documents/lfortran/src/libasr/codegen/asr_to_fortran.cpp", line 1027, in LCompilers::ASRToFortranVisitor::visit_ArrayReshape(LCompilers::ASR::ArrayReshape_t const&)
    visit_expr(*x.m_shape);
  File "/home/khushi/Documents/lfortran/src/libasr/../libasr/asr.h", line 5124, in LCompilers::ASR::BaseVisitor<LCompilers::ASRToFortranVisitor>::visit_expr(LCompilers::ASR::expr_t const&)
    void visit_expr(const expr_t &b) { visit_expr_t(b, self()); }
  File "/home/khushi/Documents/lfortran/src/libasr/../libasr/asr.h", line 4933, in ??
    case exprType::ArrayPhysicalCast: { v.visit_ArrayPhysicalCast((const ArrayPhysicalCast_t &)x); return; }
  File "/home/khushi/Documents/lfortran/src/libasr/codegen/asr_to_fortran.cpp", line 1066, in LCompilers::ASRToFortranVisitor::visit_ArrayPhysicalCast(LCompilers::ASR::ArrayPhysicalCast_t const&)
    this->visit_expr(*x.m_arg);
  File "/home/khushi/Documents/lfortran/src/libasr/../libasr/asr.h", line 5124, in LCompilers::ASR::BaseVisitor<LCompilers::ASRToFortranVisitor>::visit_expr(LCompilers::ASR::expr_t const&)
    void visit_expr(const expr_t &b) { visit_expr_t(b, self()); }
  File "/home/khushi/Documents/lfortran/src/libasr/../libasr/asr.h", line 4855, in ??
    case exprType::IntrinsicArrayFunction: { v.visit_IntrinsicArrayFunction((const IntrinsicArrayFunction_t &)x); return; }
LCompilersException: IntrinsicFunction: `Tanh` is not implemented
```

</p>
</details>

---

The problem is here: https://github.com/lfortran/lfortran/blob/22d51d770248b8e3ef5cc22642952056c18a91c5/src/libasr/codegen/asr_to_fortran.cpp#L736

The function name has to be replaced with `get_array_intrinsic_name`.
Then it prints `IntrinsicFunction: `Shape` is not implemented`
So we haven't implemented Shape yet, so the error.

---

Ah, I see. I'll add these. Thanks!